thinc/backends/numpy_ops.pyx

```diff
@@ -61,20 +61,27 @@ class NumpyOps(Ops):
             raise ValueError("BLIS support requires blis: pip install blis")
 
     def asarray(self, data, dtype=None):
+        dtype = {"dtype": dtype} if dtype is not None else {}
+
+        array = None
+        cast_array = True
+
         if isinstance(data, self.xp.ndarray):
-            if dtype is not None:
-                return self.xp.asarray(data, dtype=dtype)
-            else:
-                return self.xp.asarray(data)
+            array = self.xp.asarray(data, **dtype)
```

Comment:
Can't we just keep this as `dtype=dtype` and not create a dict at all?

Reply:
Oh wait, now I see what you're doing. Hmm...

Reply:
I'm not sure this is the worth the tradeoff for `dtype != {}` and `dtype["dtype"]`.

Reply:
Yeah, you're right - We should be able to pass `dtype=None` to `xp.asarray` without an additional check.